qiskit_machine_learning/kernels/algorithms/quantum_kernel_trainer.py

```diff
@@ -0,0 +1,225 @@
+    """Quantum Kernel Trainer Result."""
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._quantum_kernel = None  # type: QuantumKernel
+
+    @property
+    def quantum_kernel(self) -> Optional["QuantumKernel"]:
+        """Returns the optimized quantum kernel object."""
+        return self._quantum_kernel
+
+    @quantum_kernel.setter
+    def quantum_kernel(self, quantum_kernel: "QuantumKernel") -> None:
+        self._quantum_kernel = quantum_kernel
+
+
+class QuantumKernelTrainer:
+    """
+    Quantum Kernel Trainer.
+    This class provides utility to train ``QuantumKernel`` feature map parameters.
+
+    **Example**
+
+    .. code-block::
+
+        quant_kernel = QuantumKernel(
+            feature_map=...,
+            user_parameters=...,
+            quantum_instance=...
+        )
+
+        loss_func = ...
+        optimizer = ...
+        initial_point = ...
+
+        qk_trainer = QuantumKernelTrainer(
+                                        quantum_kernel=quant_kernel,
+                                        loss=loss_func,
+                                        optimizer=optimizer,
+                                        initial_point=initial_point,
+                                        )
+        qkt_results = qk_trainer.fit_kernel(X_train, y_train)
+        optimized_kernel = qkt_results.quantum_kernel
+    """
+
+    def __init__(
+        self,
+        quantum_kernel: "QuantumKernel",
+        loss: Union[str, Callable[[Sequence[float]], float]] = "svc_loss",
+        optimizer: Optimizer = SPSA(),
+        initial_point: Optional[Sequence[float]] = None,
+    ):
+        """
+        Args:
+            quantum_kernel: QuantumKernel to be trained
+            loss (Callable[[Sequence[float]], float] or str):
+                str: Loss functions available via string: {'svc_loss: SVCLoss()).
+                    If a string is passed as the loss function, then the underlying
+                    KernelLoss object will exhibit default behavior.
+                Callable[[Sequence[float]], float]: A callable loss function which takes
+                    a vector of parameter values as input and returns a loss value (float)
+            optimizer: An instance of ``Optimizer`` to be used in training. Since no
+                analytical gradient is defined for kernel loss functions, gradient-based
+                optimizers are not recommended for training kernels.
+            initial_point: Initial point from which the optimizer will begin.
+
+        Raises:
+            ValueError: unknown loss function
+        """
+        # Class fields
+        self._quantum_kernel = quantum_kernel
+        self._optimizer = optimizer
+        self._initial_point = initial_point
+        self._loss = None
+
+        # Setters
+        self.loss = loss
+
+    @property
+    def quantum_kernel(self) -> "QuantumKernel":
+        """Returns the quantum kernel object."""
```

Comment:
I would suggest replacing the word "Returns" with "Return" here.  This is based on the following guideline from [PEP 257](https://www.python.org/dev/peps/pep-0257/):

> The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".

This suggestion could also be applied to many other docstrings throughout this file.